main.py

```diff
@@ -1,16 +1,153 @@
+import asyncio
+import base64
+import datetime
+import glob
+import json
 import os
+import ssl
+from asyncio import Lock
+
+import certifi
 import decky_plugin
+from aiohttp import ClientSession
 from settings import SettingsManager
 
+
 class Plugin:
+    yt_process: asyncio.subprocess.Process | None = None
+    # We need this lock to make sure the process output isn't read by two concurrent readers at once.
+    yt_process_lock = Lock()
+    music_path = f"{decky_plugin.DECKY_PLUGIN_RUNTIME_DIR}/music"
+    cache_path = f"{decky_plugin.DECKY_PLUGIN_RUNTIME_DIR}/cache"
+    ssl_context = ssl.create_default_context(cafile=certifi.where())
+
     async def _main(self):
-         self.settings = SettingsManager(name="config", settings_directory=decky_plugin.DECKY_PLUGIN_SETTINGS_DIR)
+        self.settings = SettingsManager(
+            name="config", settings_directory=decky_plugin.DECKY_PLUGIN_SETTINGS_DIR
+        )
 
     async def _unload(self):
-        pass
+        if self.yt_process is not None:
+            self.yt_process.terminate()
```

Comment:
Add `await self.yt_process.wait()` to wait for process to terminate.

Reply:
Do we really need to wait for it to terminate on unload, though? It should be enough to just issue the SIGTERM and then just let it terminate in the background, right?

Reply:
SIGTERM can be ignored by the process, so it is better to either wait for the process to end with some sort of timeout with a later SIGKILL if it doesn't end in time. A process can be left behind in a running state, which we don't be able to kill without a reboot if we don't ensure of the process being ended.

Reply:
That's a good point. I changed it in 0e4680c to send SIGKILL if SIGTERM is stuck for more than 5 seconds.